raft/fsm.cc

```diff
@@ -19,9 +19,10 @@ leader::~leader() {
 }
 
 fsm::fsm(server_id id, term_t current_term, server_id voted_for, log log,
-        index_t commit_idx, failure_detector& failure_detector, fsm_config config) :
+        index_t commit_idx, failure_detector& failure_detector, fsm_config config,
+        seastar::condition_variable& sm_events) :
```

Comment:
We can abstract it further by providing `void signal_even()` callback

Reply:
Do you prefer the further abstraction? I have no preference, I can do it if you really want. (Would rather not if there's no practical reason though)

Reply:
I would still need to add a constructor to `fsm_debug` which would provide a no-op `signal_event`. If you make the counter-proposal of providing a default value of this parameter, I am strongly against it as default parameters values are traps, we had to fix multiple bugs thanks to them already. (E.g. the recent abort_source one fixed by Patryk)

Reply:
Or you can just pass `[]()` in tests.

Reply:
Imo better to extend `fsm_debug` to not have to modify each of the tens of use sites in tests.

Reply:
I am fine with adding a constructor to fsm_debug that will set it to empty as well. I fine with default parameters as well for the record :)

Reply:
I decided to leave the condition_variable&, I think it's a bit simpler.